Documentation/proposals/202403-prometheus-operator-cli.md

```diff
@@ -0,0 +1,92 @@
+# Prometheus Operator CLI
+
+- Owners:
+  - [Nicolas Takashi](https://github.com/nicolastakashi)
+- Related Tickets
+  - [#6423](https://github.com/prometheus-operator/prometheus-operator/issues/6423)
+- Other docs:
+  - N/A
+
+This document proposes a new CLI tool to manage Prometheus Operator resources, allowing users to create, update, and delete Prometheus Operator resources using a simple CLI interface as well as
+troubleshoot and debug Prometheus Operator resources.
+
+# Why
+
+During KubeCon Paris 2024, we received feedback from users that managing Prometheus Operator resources is difficult and error-prone. Users reported that they often encounter issues when creating, updating, and deleting Prometheus Operator resources, and that troubleshooting and debugging Prometheus Operator resources is challenging, below you can find some of the issues that users reported:
+
+## Why my Prometheus resource is not being created?
+
+People were struggling to troubleshoot why their Prometheus resources were not being created. They were not sure if the issue was with the Prometheus Operator or with the Prometheus resource itself.
+After long investigation, they found out that the Prometheus Operator were only watching the namespace where the Prometheus Operator was deployed, and the Prometheus resource was being created in a different namespace.
+
+## Why my ServiceMonitor is not being created?
+
+People were struggling to troubleshoot why their ServiceMonitor resources were not being created. They were not sure if the issue was with the Prometheus Operator or with the ServiceMonitor resource itself.
+
+After long investigation, they found out that the matchingLabels field in the ServiceMonitor resource was not matching the labels of the target Service.
+
+These are just some common issues that users reported, but there are many other issues that users face when managing Prometheus Operator resources, and we believe that a CLI tool can help users to manage Prometheus Operator resources more easily and efficiently.
+
+# Pitfalls of the current solution
+
+At the moment people are struggling to manage Prometheus Operator resources, they have to manually create, update, and delete Prometheus Operator resources using `kubectl` or other tools, and troubleshooting and debugging Prometheus Operator resources is challenging.
```

Comment:
I think the issues you pointed out above can be explored and explained in this section instead of having subsections under `Why`.

Troubleshooting Alertmanager, Prometheus and Thanos ruler pod creation is very manual and requires domain knowledge about what RBAC permissions are needed.

Target discovery also requires domain knowledge, understanding label selectors, RBAC permissions, etc. 

And we could add links to existing support issues here in Github, StackOverflow, or Slack to show that we have data to back this up.

Reply:
And what do you think about not adding a subsection for the problems mentioned? I feel like it's an overkill to add very big title just to say 1 paragraph 🤔

Reply:
@ArthurSens could you review?